File: solutions/go/hamming/2/hamming.go
package hamming

import "errors"

func Distance(a, b string) (distance int, err error) {
    if len(a) != len(b) {
        err = errors.New("length is different")
        return
    }

    for i := 0; i < len(a); i += 1 {
        if a[i] != b[i] {
            distance += 1
        }
    }
    return
}

File: solutions/python/hamming/2/hamming.py
def distance(strand_a: str, strand_b: str) -> int:
    """
    Calculating the Hamming Distance between two DNA strands.
    """
    if not len(strand_a) - len(strand_b):
        return sum(a != b for a, b in zip(strand_a, strand_b))

    raise ValueError("Strands must be of equal length.")
